plugin/pkg/scheduler/factory/factory.go

```diff
@@ -478,48 +478,48 @@ func getNodeConditionPredicate() cache.NodeConditionPredicate {
 // scheduled.
 func (factory *ConfigFactory) createUnassignedNonTerminatedPodLW() *cache.ListWatch {
 	selector := fields.ParseSelectorOrDie("spec.nodeName==" + "" + ",status.phase!=" + string(api.PodSucceeded) + ",status.phase!=" + string(api.PodFailed))
-	return cache.NewListWatchFromClient(factory.Client, "pods", api.NamespaceAll, selector)
+	return cache.NewListWatchFromClient(factory.Client.Core().GetRESTClient(), "pods", api.NamespaceAll, selector)
```

Comment:
I agree with that. Though, scheduler is not the only occurrence of the phenomenon. There are other places in kubelet, apiserver, proxy, etc. that are subjects to the replacement. I would suggest to open separate PR for it.